beamer/chains.py

```diff
@@ -0,0 +1,106 @@
+from dataclasses import dataclass
+from beamer.typing import ChainId
+
+
+@dataclass(frozen=True)
+class ChainDescriptor:
+    id: ChainId
+    ecosystem_name: str
+    name: str
+    type2: bool = False
+    bedrock: bool = False
+    testnet: bool = False
+    local: bool = False
+
+    @property
+    def supports_type_2_transactions(self) -> bool:
+        return self.type2 and not self.local
+
+    @property
+    def is_testnet(self) -> bool:
+        return self.testnet
+
+    @property
+    def is_bedrock(self) -> bool:
+        return self.bedrock
+
+
+@dataclass(frozen=True)
+class _BeamerConfig:
+    chains: dict[str, dict[str, ChainDescriptor]]
+
+    def get_descriptor(self, chain_id: ChainId) -> ChainDescriptor:
+        for networks in self.chains.values():
+            for descriptor in networks.values():
+                if descriptor.id == chain_id:
+                    return descriptor
+        return ChainDescriptor(id=chain_id, ecosystem_name="default", name="default")
```

Comment:
We don't really need the `_BeamerConfig` type, it does not do much anyway.

Reply:
Regarding the return value of `get_descriptor`, we have two options if the chain ID is not found:

1. returning `None`
2. returning an instance of `ChainDescriptor` that somehow indicates that this chain is not known to us

I think the agent actually drops things that are coming from unknown chains so I _think_ by the time this function is called, it should be the case that the chain is known. However, this module lives at `beamer.chains` and is expected to be used in a wider context i.e. not just the agent. So I think we should actaully return `None` here.

In fact,
```python
if get_chain_descriptor(chain_id) is not None:
```
can server as the `is this chain supported/know by us` check.

Reply:
One other thing here: I think it would be nice if, during execution of this module, the toplevel data that contains the descriptors were transformed in such a way so that it makes descriptor lookup O(1) instead of O(n).